src/main/scala/api/other/RegistrationComparison.scala

```diff
@@ -26,4 +27,25 @@ object RegistrationComparison {
     val hausdorffDistance = MeshMetrics.hausdorffDistance(reconstruction, groundTruth)
     println(s"ID: ${id} average2surface: ${avgDist2Surf} hausdorff: ${hausdorffDistance}")
   }
+
+  private def avgDistanceBoundaryAware(m1: TriangleMesh[_3D], m2: TriangleMesh[_3D]): (Double, Double) = {
+
+    val pointsOnSample = m1.pointSet.points
+    val dists = for (p <- pointsOnSample) yield {
+      val pTarget = m2.operations.closestPointOnSurface(p).point
+      val pTargetId = m2.pointSet.findClosestPoint(pTarget).id
+      if (m2.operations.pointIsOnBoundary(pTargetId)) -1.0
```

Comment:
I would use an Option instead of -1 to show that no value could be produced. This makes intention more clear